### PR TITLE
v1.9 backports 2021-04-09

### PR DESCRIPTION
**Pull request by errordeveloper**
* #15453 -- eni: Assign primary IP to support multiple VPC CIDRs (@gandro)
 * #15261 -- node-neigh: Reduce arping related log msg's level (@brb)
 * #15598 -- Make encryption+chaining limitations clearer (@joestringer)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 15453 15261 15598; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Comment by errordeveloper** [comment]
test-backport-1.9

**Review by joestringer — APPROVED** [review]

**Review by brb — CHANGES_REQUESTED** [review]
My changes depend on https://github.com/cilium/cilium/pull/15536 backported first.

File: pkg/aws/ec2/ec2.go [hunk 1]
```diff
@@ -195,8 +195,8 @@ func parseENI(iface *ec2.NetworkInterface, vpcs ipamTypes.VirtualNetworkMap, sub
 	}
 
 	for _, ip := range iface.PrivateIpAddresses {
-		if ip.PrivateIpAddress != nil {
-			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
+		if ip.PrivateIpAddress != nil && !ip.Primary {
+			eni.Addresses = append(eni.Addresses, aws.ToString(ip.PrivateIpAddress))
```
Comment (errordeveloper) [inline pkg/aws/ec2/ec2.go]: @gandro do these lines need changing or okay like this for 1.9?

Comment (gandro) [reply pkg/aws/ec2/ec2.go]: As discussed offline:
```suggestion
		if ip.PrivateIpAddress != nil && (ip.Primary == nil || !*ip.Primary) {
			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
```

**Comment by errordeveloper** [comment]
@brb I've rebased this on https://github.com/cilium/cilium/pull/15536 - PTAL :)

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -727,7 +726,8 @@ func (n *linuxNodeHandler) insertNeighbor(ctx context.Context, newNode *nodeType
 	if nextHopIsNew || refresh {
 		hwAddr, _, err := arping.PingOverIface(nextHopIPv4, n.neighDiscoveryLink, srcIPv4)
 		if err != nil {
-			scopedLog.WithError(err).Error("arping failed")
+			scopedLog.WithError(err).Info("arping failed")
+			metrics.ArpingRequestsTotal.WithLabelValues(failed).Inc()
```
Comment (errordeveloper) [inline pkg/datapath/linux/node.go]: @brb looks like this drags in 42eb1edafdaccad2814e7c16a1795e501fcd76b3 as well, which does not apply cleanly, I suppose I should just drop this line from the backport?

Comment (errordeveloper) [reply pkg/datapath/linux/node.go]: Also adding metrics is actually feature, albeit a small one...

Comment (brb) [reply pkg/datapath/linux/node.go]: The upstream commit https://github.com/cilium/cilium/commit/2de65d0e77df7d958ace1839c595261877e4988d does not include the metric change. Did it get by accident?

Comment (errordeveloper) [reply pkg/datapath/linux/node.go]: Ah, I see, it must have slipped when I was resolving the conflict...

**Comment by errordeveloper** [comment]
@brb I've dropped reference to `metrics` - PTAL :)

**Review by gandro — APPROVED** [review]
Looks good for my commits!

**Review by brb — APPROVED** [review]
My changes LGTM, thanks!

**Comment by errordeveloper** [comment]
test-backport-1.9

**Review by qmonnet — APPROVED** [review]

**Pull request merged**